planetiler-core/src/main/java/com/onthegomap/planetiler/geo/TileCoord.java

```diff
@@ -101,11 +109,16 @@ public String toString() {
     return "{x=" + x + " y=" + y + " z=" + z + '}';
   }
 
-  public double progressOnLevel(TileExtents extents) {
-    // approximate percent complete within a bounding box by computing what % of the way through the columns we are
-    // (for hilbert ordering, we probably won't be able to reflect the bounding box)
-    var zoomBounds = extents.getForZoom(z);
-    return 1d * (x - zoomBounds.minX()) / (zoomBounds.maxX() - zoomBounds.minX());
+  public double progressOnLevel(TileOrder order, TileExtents extents) {
+    if (order == TileOrder.TMS) {
+      // approximate percent complete within a bounding box by computing what % of the way through the columns we are
+      var zoomBounds = extents.getForZoom(z);
+      return 1d * (x - zoomBounds.minX()) / (zoomBounds.maxX() - zoomBounds.minX());
+    } else {
+      // this assumes extents is the whole world, so will be wrong when tiling a limited bbox
+      // we could improve this by making TileExtents store an exact set of all encoded coordinates.
+      return 1d * Hilbert.hilbertXYToIndex(this.z, this.x, this.y) / (2 << this.z);
```

Comment:
Shouldn't this be `(1 << (2 * this.z))` to get 1, 4, 16, 64, ... for the levels?